base/message_loop/message_pump_io_starboard.cc

```diff
@@ -52,7 +52,14 @@ bool MessagePumpIOStarboard::SocketWatcher::StopWatchingSocket() {
   bool result = true;
   if (SbSocketIsValid(socket)) {
     DCHECK(pump_);
+#if defined(STARBOARD)
```

Comment:
Does the DCHECK above still needed? 
Should we gate the code the code with a mutex instead of #if defined(STARBOARD)?

Reply:
It would be nice to get an assert if this ever reproduces on a devel/debug build because it may help with a full root cause. For qa and gold builds, DCHECKs are no-ops.

Reply:
How about using mutex, like this?

  {
    if (SbSocketIsValid(socket)) {
      ScopedLock lock(mutex_);
      if (pump_) {
        result = pump_->StopWatching(socket);
      }
      pump_ = nullptr;
    }
  }
  return result;

Reply:
This isn't being called from separate threads (if it is, then that is a different bug). It's getting called repeatedly from the same thread repeatedly. There is nothing to synchronize here for a mutex.

Reply:
should we remove the macro?
#if defined(STARBOARD)

Reply:
We should gate all our changes that we make to upstream code, because that clarifies that we rely on it so that the change doesn't get removed during manual conflict resolution when we rebase.